### PR TITLE
Replace App Engine users module with Firebase Auth (#399).

### PR DESCRIPTION
**Pull request by oliverchang**
- Implement our own libs.auth module, which provide similar
  functionality to the App Engine users module.
- Add a login page which uses
  https://github.com/firebase/firebaseui-web. This gives us a token that
  we store as a cookie and validate on every request.
- Add a sync-admins cron job to periodically update the list of admin
  users.
- Update docs and create_config to include firebase setup.

### CODE REVIEW

File: src/python/tests/appengine/handlers/cron/sync_admins_test.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                    'user:user3@email.com',
+                ]
+            },
+            {
+                'role': 'roles/owner',
+                'members': ['user:user4@email.com',]
+            },
+            {
+                'role': 'roles/viewer',
+                'members': ['user:user5@email.com',]
```
Comment (inferno-chromium) [inline src/python/tests/appengine/handlers/cron/sync_admins_test.py]: Should viewer be admins ?

File: src/appengine/handlers/cron/sync_admins.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    existing_admin_emails.add(admin.email)
+
+  ndb.delete_multi(to_remove)
+
+  to_add = []
+  for admin in new_admins:
+    if admin not in existing_admin_emails:
+      to_add.append(data_types.Admin(id=admin, email=admin))
+
```
Comment (inferno-chromium) [inline src/appengine/handlers/cron/sync_admins.py]: maybe add some logs.log to tell which users are getting added as admins. also, look for any places that need more logging for better future debugging.

**Comment by inferno-chromium** [comment]
2 more small things when looking at ui, can be done in next cl
1. we show please login page first and then when we click it, we show the firebase login page. we can just remove the first page now, it feels redundant since only way to move forward to go to login page. we should just redirect there.
2. login button is somewhat misaligned with login in title bar.

File: docs/production-setup/clusterfuzz.md [hunk 3]
```diff
@@ -41,6 +41,21 @@ If you're new to Google Cloud you may be eligible for [trial credit].
 
 [trial credit]: https://cloud.google.com/free/docs/gcp-free-tier#free-trial
 
+## Enable Firebase
+Follow [these instructions](https://cloud.google.com/appengine/docs/standard/python3/building-app/adding-firebase)
+to add Firebase to the Google Cloud project you just created. **Important**: You
```
[the remark this answers is not in the document]
Comment (oliverchang) [reply docs/production-setup/clusterfuzz.md]: Done!

[the remark this answers is not in the document]
Comment (oliverchang) [reply docs/production-setup/clusterfuzz.md]: Done!

File: docs/production-setup/clusterfuzz.md [hunk 4]
```diff
@@ -67,7 +82,7 @@ settings for your deployment and can be later updated.
 ```bash
 mkdir /path/to/myconfig  # Any EMPTY directory outside the ClusterFuzz source repository.
 export CONFIG_DIR=/path/to/myconfig
-python butler.py create_config --oauth-client-secrets-path=$CLIENT_SECRETS_PATH --project-id=$CLOUD_PROJECT_ID $CONFIG_DIR
+python butler.py create_config --oauth-client-secrets-path=$CLIENT_SECRETS_PATH --firebase-api-key=$FIREBASE_API_KEY --project-id=$CLOUD_PROJECT_ID $CONFIG_DIR
```
[the remark this answers is not in the document]
Comment (oliverchang) [reply docs/production-setup/clusterfuzz.md]: Done!

**Pull request merged**